### PR TITLE
feat(internal/request/http.go): check impersonation also of groups

### PR DESCRIPTION
**Pull request by maxgio92**
Signed-off-by: Massimiliano Giovagnoli <me@maxgio.it>

### CODE REVIEW

**Review by prometherion — COMMENTED** [review]
I'm not aware of RBAC regarding impersonating groups in Kubernetes: could you point to the documentation about it?

Just saying that I always saw the verb `impersonate` about users, and not groups.

File: internal/request/http.go [hunk 1]
```diff
@@ -176,3 +204,12 @@ func (h http) isJwtToken() bool {
 
 	return err == nil
 }
+
+func contains(ss []string, s string) bool {
```
Comment (prometherion) [inline internal/request/http.go:208]: You can use the `sets.NewString` utility to check if an array contains a value. It's the same logarithmic complexity, it's much more readable.

**Review by prometherion — CHANGES_REQUESTED** [review]

**Comment by maxgio92** [comment]
> I'm not aware of RBAC regarding impersonating groups in Kubernetes: could you point to the documentation about it?
> 
> Just saying that I always saw the verb `impersonate` about users, and not groups.

Yes, [here](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation). AFAIU a `ClusterRole` rule on `impersonate` verb can apply to `["users", "groups", "serviceaccounts"]` resources.

File: internal/request/http.go (hunk 1 above)
Comment (maxgio92) [reply internal/request/http.go]: Applied @prometherion, thanks for pointing me out it

**Pull request merged**